datafusion/functions-array/src/string.rs

```diff
@@ -281,6 +281,31 @@ pub(super) fn array_to_string_inner(args: &[ArrayRef]) -> Result<ArrayRef> {
 
                 Ok(arg)
             }
+            Dictionary(..) => {
+                let any_dict_array = arr
+                    .as_any_dictionary_opt()
+                    .ok_or_else( || {
+                        DataFusionError::Internal(
+                            format!("could not cast {} to AnyDictionaryArray", arr.data_type())
+                        )
+                    })?;
+                macro_rules! array_function {
+                    ($ARRAY_TYPE:ident) => {
+                        to_string!(
+                            arg,
+                            any_dict_array.values(),
+                            &delimiter,
+                            &null_string,
+                            with_null_string,
+                            $ARRAY_TYPE
+                        )
+                    };
+                }
+                call_array_function!(
+                    any_dict_array.values().data_type(),
+                    false
+                )
```

Comment:
Why not use
```suggestion
                compute_array_to_string(
                    arg,
                    any_dict_array.values().clone(),
                    delimiter.clone(),
                    null_string.clone(),
                    with_null_string,
                )?;

                Ok(arg)
```

Reply:
I wanted to see if I could avoid calling again `compute_array_to_string` and instead compute the result and return inmediately, but calling `compute_array_to_string` is what other data types are doing, so makes sense to follow the same pattern.